cmd/dagger/module.go

```diff
@@ -230,6 +231,60 @@ var moduleInstallCmd = &cobra.Command{
 			if err != nil {
 				return fmt.Errorf("failed to generate code: %w", err)
 			}
+
```

Comment:
@vito @sipsma Can you double check this here?

Reply:
There's a small subtly here, I think at the moment the code below will likely error out when calling `depSrc.ModuleName` and `depSrc.AsModule` in the case where the dep is local (git is fine though).

The reason being that local dependencies are provided as pure relative paths, no filesystem associated yet. The engine internally does the path arithmetic of making that path relative to the parent module. That's the point at which those APIs can succeed (since they have a filesystem to read config+source from).

That API for resolving deps relative to parents is actually available to SDKs though too, so you should just be able to do:
```go
resolvedDep := modConf.Mod.Source().ResolveDependency(depSrc)

name, err := resolvedDep.Source().ModuleName(ctx)
// etc.
```

Reply:
non-blocking: it's now possible to install a dependency with a custom name (`installName` var used above), which may be something worth tossing into the analytics. Not incredibly important to track either though, so no big deal either way.